Core/include/Acts/Propagator/ActorConcepts.hpp

```diff
@@ -0,0 +1,105 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2024 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include <concepts>
+#include <type_traits>
+#include <utility>
+
+namespace Acts {
+
+template <typename actor_t>
+concept ActorHasResult = requires { typename actor_t::result_type; };
+
+template <typename actor_t, typename propagator_state_t, typename stepper_t,
+          typename navigator_t, typename... Args>
+concept ActorHasActWithResult =
+    requires(const actor_t& a, propagator_state_t& state,
+             const stepper_t& stepper, const navigator_t& navigator,
+             typename actor_t::result_type& result, Args&&... args) {
+      {
+        a.act(state, stepper, navigator, result, std::forward<Args>(args)...)
+      } -> std::same_as<void>;
+    };
```

Comment:
I think it would make sense to require `ActorHasResult` here:

```suggestion
concept ActorHasActWithResult = ActorHasResultr<T> &&
    requires(const actor_t& a, propagator_state_t& state,
             const stepper_t& stepper, const navigator_t& navigator,
             typename actor_t::result_type& result, Args&&... args) {
      {
        a.act(state, stepper, navigator, result, std::forward<Args>(args)...)
      } -> std::same_as<void>;
    };
```